did-pkh-method-draft.md

```diff
@@ -0,0 +1,256 @@
+all did-pkh addresses; in blockchain systems where accounts are controlled by
+multiple keytypes, like Tezos, the network and chain subdomains will not be
+enough to identify keytype, which must be detected from the address itself.
+
+|account type|network id (CAIP-2) + chain id (CAIP-10)|verification method type|URL for context definition|
+|---|---|---|---|
+|`tz1`|`tezos:NetXdQprcVkpaWU`|Ed25519PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021|https://w3id.org/security#Ed25519PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021|
+|`tz`|`tezos:NetXdQprcVkpaWU`|EcdsaSecp256k1RecoveryMethod2020|https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020|
+|`tz`|`tezos:NetXdQprcVkpaWU`|P256PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021|https://w3id.org/security#P256PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021|
+|ethereum mainnet|`eip155:1`|EcdsaSecp256k1RecoveryMethod2020|https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020|
+|celo mainnet|`eip155:42220`|EcdsaSecp256k1RecoveryMethod2020|https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020|
+|polygon mainnet|`eip155:137`|EcdsaSecp256k1RecoveryMethod2020|https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020|
+|solana|`solana:4sGjMW1sUnHzSxGspuhpqLDx6wiyjNtZ`|Ed25519VerificationKey2018|https://w3id.org/security#Ed25519VerificationKey2018|
+|bitcoin mainnet|`bip122:000000000019d6689c085ae165831e93`|EcdsaSecp256k1RecoveryMethod2020|https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020|
+|dogecoin mainnet|`bip122:1a91e3dace36e2be3bf030a65679fe82`|EcdsaSecp256k1RecoveryMethod2020|https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020|
+
+### Context
+
+The following should be manually inserted into each DID Document. This will
+likely change over time as new verification methods are supported, and
+general-purpose methods are specified. Term definitions may be omitted from
+these if they are not needed in particular DID documents.
+
+```
+{
+  "blockchainAccountId": "https://w3id.org/security#blockchainAccountId",
+  "publicKeyJwk": {
+    "@id": "https://w3id.org/security#publicKeyJwk",
+    "@type": "@json"
+  },
+  "Ed25519VerificationKey2018": "https://w3id.org/security#Ed25519VerificationKey2018",
+  "Ed25519PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021": "https://w3id.org/security#Ed25519PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021",
+  "P256PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021": "https://w3id.org/security#P256PublicKeyBLAKE2BDigestSize20Base58CheckEncoded2021",
+  "TezosMethod2021": "https://w3id.org/security#TezosMethod2021",
+  "EcdsaSecp256k1RecoveryMethod2020": "https://identity.foundation/EcdsaSecp256k1RecoverySignature2020#EcdsaSecp256k1RecoveryMethod2020"
+}
```

Comment:
Should all of these really be included in each DID doc? I assume you can select specific ones based on which chain is used?

Reply:
>  Term definitions may be omitted from these if they are not needed in particular DID documents.

maybe this could be made more explicit, since someone who doesn't understand @Context might find it nebulous to guess which are "needed".  i.e. 

> The subset of the following terms that are used by a given PKH system should be defined in the @Context field of each DID Document derived from a PKH; the rest can be omitted.

Reply:
The last 5 term definitions in this context object are for verification method types. Which verification method to use per chain (or chain + address prefix in the case of Tezos) can be determined using the Networks table (verification method type → URL for context definition). Whether or not to use the other two definitions, `publicKeyJwk` or `blockchainAccountId`, depends on the verification method type. We haven't specified that yet (except in this test vectors), but basically if the DID encodes the actual public key, `publicKeyJwk` can be used (decoding/translating the address as appropriate), otherwise `blockchainAccountId` must be used (passing through the CAIP-10 account id.

An alternative to constructing the context object inline would be to use a remote context file - either one for the current object, or specifying a URI per chain type.

Another idea would be use a single verification method for all types; [BlockchainVerificationMethod2021](https://or13.github.io/lds-blockchain2021/#BlockchainVerificationMethod2021) looks like a good candidate, as it is based on `blockchainAccountId`. (We may need changes in it for Tezos tz1/tz3: https://github.com/OR13/lds-blockchain2021/issues/1). But using more wider-supported verification method types where possible (e.g. `EcdsaSecp256k1RecoveryMethod2020`, `Ed25519VerificationKey2018`, is better for interoperability.